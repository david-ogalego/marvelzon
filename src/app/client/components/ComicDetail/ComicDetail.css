.containerComicDetail {
    float: left;
    margin: 50px 30px 20px 30px;
}

.imageContainer {
    min-width: 550px;
}

.containerBasicDetail {
    display: flex;
    flex-direction: row;
}

.detailsContainer {
    display: flex;
    flex-direction: column;
    padding-left: 10px;
}