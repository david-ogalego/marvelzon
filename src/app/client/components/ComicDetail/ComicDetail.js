import React from 'react';
import stylesComicDetail from './ComicDetail.css';
import { getDateFormated } from '../../helpers/date';

const ComicDetail = function(props) {
    return (
        <div className={stylesComicDetail.containerComicDetail}>
            <h1>{props.title}</h1>
            <div className={stylesComicDetail.containerBasicDetail}>
                <div className={stylesComicDetail.imageContainer}>
                    <img alt={props.title} src={props.image} />
                </div>
                <div className={stylesComicDetail.detailsContainer}>
                    <label>{props.description}</label>
                    <label>Date: {getDateFormated(props.date)}</label>
                    <label>Price: {props.price}</label>
                    <label>Number pages: {props.pageCount}</label>
                    <a href={props.url} target="_blank" >Detail</a>
                </div>
            </div>
        </div>
    );
}

export default ComicDetail;