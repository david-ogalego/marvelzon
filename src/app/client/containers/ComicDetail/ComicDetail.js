import React from 'react';

const ComicDetail = function({ match }) {
    return (
        <h1>Detail {match.params.comicId}</h1>
        
    );
}

export default ComicDetail;