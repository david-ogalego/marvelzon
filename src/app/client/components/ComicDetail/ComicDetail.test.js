import React from 'react';
import renderer from 'react-test-renderer';
import ComicDetail from './ComicDetail';
import MarvelLogo from '../../images/MarvelLogo.png';

test('Snapshot ComicDetail rendered with comic', () => {
  const treeComicDetail = renderer.create(<ComicDetail
    title="X-Men"
    date="2029-12-31T00:00:00-0500"
    price="200"
    image={MarvelLogo}
    pageCount={10}
    description="Lorem Ipsum ha sido el texto de relleno estÃ¡ndar"
    url="http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b"
  />).toJSON();
  expect(treeComicDetail).toMatchSnapshot();
});
