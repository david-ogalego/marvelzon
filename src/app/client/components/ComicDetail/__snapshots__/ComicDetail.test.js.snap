// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ComicDetail rendered with comic 1`] = `
<div
  className="containerComicDetail"
>
  <h1>
    X-Men
  </h1>
  <div
    className="containerBasicDetail"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="X-Men"
        src="test-file-stub"
      />
    </div>
    <div
      className="detailsContainer"
    >
      <label>
        Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen.
      </label>
      <label>
        Date: 
        2029-12-31
      </label>
      <label>
        Price: 
        200
      </label>
      <label>
        Number pages: 
        10
      </label>
      <a
        href="http://marvel.com/comics/issue/2088/official_handbook_of_the_marvel_universe_2004_14_fantastic_four/fantastic_four?utm_campaign=apiRef&utm_source=d86beaee5f52cf5b1205630a7e35b24b"
        target="_blank"
      >
        Detail
      </a>
    </div>
  </div>
</div>
`;
